packages/react/src/hooks/actions/useDataStoreCreateAction.ts

```diff
@@ -13,42 +14,69 @@ import {
 } from './constants';
 import { getErrorMessage } from '../../helpers/utils';
 import { AMPLIFY_SYMBOL } from '../../helpers/constants';
+import { useTypeCastFields } from './shared/useTypeCastFields';
+import { ModelFormFields } from './shared/typeUtils';
 
 export interface UseDataStoreCreateActionOptions<
   Model extends PersistentModel
 > {
+  /**
+   * Expected datastore model
+   */
   model: PersistentModelConstructor<Model>;
-  fields: ModelInit<Model, { readOnlyFields: 'createdAt' | 'updatedAt' }>;
+  /**
+   * Strongly typed fields based on model. Ensure fields are already
+   * the appropriate type before passing to fields.
+   */
+  fields?: ModelInit<Model>;
+  /**
+   * Used in place of `fields` in combination with `schema` param to have
+   * string field values optimistically cast to the expected type based on the schema
+   */
+  formFields?: ModelFormFields<ModelInit<Model>>;
+  /**
+   * Used in combination with `formFields` param for string field values
+   * optimistically cast to the expected type based on the schema
+   */
+  schema?: Schema;
 }
 
 /**
  * Action to Create DataStore item
  * @internal
  */
-export const useDataStoreCreateAction =
-  <Model extends PersistentModel>({
-    model,
+export const useDataStoreCreateAction = <Model extends PersistentModel>({
+  model,
+  fields,
+  formFields,
+  schema,
+}: UseDataStoreCreateActionOptions<Model>) => {
+  const fieldsOrConvertedFields = useTypeCastFields<Model>({
```

Comment:
Nit: `fieldsOrConvertedFields` reads a bit off to me since this function doesn't care whether the fields are converted or not, maybe just name `fields` and reassign the `fields` key on ln 51 to `initialFields`?

Reply:
Love it, will update